bridge/src/lib.rs

```diff
@@ -464,16 +469,26 @@ pub mod pallet {
 			// parse proposals and construct signing message
 			let final_message = Self::construct_ecdsa_signing_proposals_data(proposals);
 
-			// recover the signing pubkey
+			// recover the signing address
 			if let Ok(pubkey) =
-				secp256k1_ecdsa_recover_compressed(&sig, &blake2_256(&final_message))
+				// recover the uncompressed pubkey
+				secp256k1_ecdsa_recover(&sig, &blake2_256(&final_message))
 			{
-				pubkey == MpcKey::<T>::get().0
+				let address = Self::public_key_to_address(&pubkey);
+
+				address == MpcAdd::<T>::get().0
 			} else {
 				false
 			}
 		}
 
+		/// convert the ECDSA 64-byte uncompressed pubkey to H160 address
+		pub fn public_key_to_address(public_key: &[u8]) -> [u8; 20] {
+			let hash = keccak_256(public_key);
+			let final_hash = array_ref![&hash, 12, 20];
```

Comment:
Oh, I don't know the function of [12, 20] here, can we add some comments?

Reply:
I figure it out. It's part of the Ethereum address generation

Reply:
yes, uncompressed pubkey is a 64 bytes array, then we hash it with keccak256 to get a 32 bytes hash, the last 20 bytes will be the H160 address